cmd/auth-rpc-client.go

```diff
@@ -182,3 +194,76 @@ func (authClient *AuthRPCClient) ServerAddr() string {
 func (authClient *AuthRPCClient) ServiceEndpoint() string {
 	return authClient.config.serviceEndpoint
 }
+
+// defaultDialTimeout is used for non-secure connection.
+const defaultDialTimeout = 3 * time.Second
+
+// dial tries to establish a connection to serverAddr in a safe manner.
+// If there is a valid rpc.Cliemt, it returns that else creates a new one.
+func rpcDial(serverAddr, serviceEndpoint string, secureConn bool) (netRPCClient *rpc.Client, err error) {
+	var conn net.Conn
+	if secureConn {
+		var hostname string
+		if hostname, _, err = net.SplitHostPort(serverAddr); err != nil {
+			err = &net.OpError{
+				Op:   "dial-http",
+				Net:  serverAddr + serviceEndpoint,
+				Addr: nil,
+				Err:  fmt.Errorf("Unable to parse server address <%s>: %s", serverAddr, err.Error()),
+			}
+
+			return nil, err
+		}
+
+		// ServerName in tls.Config needs to be specified to support SNI certificates.
+		conn, err = tls.Dial("tcp", serverAddr, &tls.Config{ServerName: hostname, RootCAs: globalRootCAs})
```

Comment:
It is also possible to add a timeout for tls.Dial, this way:
```
d := &net.Dialer{
	Timeout:   defaultDialTimeout,
}
```
and
```
conn, err = tls.DialWithDialer(d, "tcp", ...)
```

also 
```
conn.err = net.DialTimeout("tcp", serverAddr, defaultTimeout)
```

can be replaced with:
```
d.Dial("tcp", "serverAddr")
```

Reply:
Nice thanks @vadmeste good change.